src/components/state-details.js

```diff
@@ -94,28 +94,251 @@ const getPlacesData = (data) => {
   return allPlacesData
 }
 
+const currentYear = new Date().getFullYear()
+
+// We want to get to 0 by 2050 and we use our current emissions as a start,
+// so the % to cut by is 100 divided by the number of years we have
+const cutPerYearPrcnt = (100 / (2050 - currentYear)).toFixed(1)
+
 const StateDetailsPage = ({location, data}) => {
-  const emissions_2018 = get_2018_emissions(data)
-  console.log('location', location)
   const currentPlace = location.pathname.split("/")[1]
-
+  const countryEmissions = get_2018_emissions(data)
   const placesData = getPlacesData(data)
-  const emissions= emissions_2018
 
-  const [placeData, setPlaceData] = useState(placesData[currentPlace])
+  const currPlaceData = placesData[currentPlace]
+
+  // Get the last year of emissions data we have to use for showing the
+  // breakdown of how much emission comes from each source in this state
+  const placeEmissions = currPlaceData.emissions[currPlaceData.emissions.length - 1]
+
+  const totalLatestEmissions = placeEmissions.buildings +
+    placeEmissions.dirty_power +
+    placeEmissions.dumps_farms_industrial_other +
+    placeEmissions.transportation
+
+  const buildingsPrcnt = (placeEmissions.buildings / totalLatestEmissions * 100).toFixed(1)
+  const powerPrcnt = (placeEmissions.dirty_power / totalLatestEmissions * 100).toFixed(1)
+  const transportPrcnt = (placeEmissions.transportation / totalLatestEmissions * 100).toFixed(1)
+
+  const placeTitle = currPlaceData.name
+
+  const [placeData, setPlaceData] = useState(currPlaceData)
 
   return (
     <Layout>
       <SEO />
 
-      <div className='row d-flex flex-row'>
-        <div className='col-12 col-lg-4'>
-          <h1 className='mr-4 mb-3'>{placeData.name}</h1>
-          <ChoroplethMap emissions={emissions} sidebar={false} selected_location={currentPlace}/>
+      <div className='col-6'>
```

Comment:
There is a LOT of HTML in this one component, so if it makes sense to break it up (maybe into a power, building, and transport section or something) let me know. I also don't like the duplication of all of these Bootstrap HTML classes, so it might make sense to just make something like `p className="details-content"` and then apply the standard styles to that.

Reply:
i'd be in favor of breaking things up if we planned on reusing them in other places, but right now the state detail page is the only place it will be. it may be a bit long, but one page/file makes more intuitive sense to me.

for bootstrap classes, yes they're a bit redundant, but i dont want to go down the path of writing our own css unless we need to. we can get 95% of what we need from applying the standard classes appropriately.